collect_app/src/test/java/org/odk/collect/android/formmanagement/ServerFormsDetailsFetcherTest.java

```diff
@@ -131,13 +131,48 @@ public void whenAFormExists_andHasUpdatedMediaFileOnServer_isUpdated() throws Ex
         assertThat(getForm(serverFormDetails, "form-2").isUpdated(), is(true));
     }
 
+    @Test
+    public void whenAFormExists_andItsNewerVersionWithUpdatedMediaFilesHasBeenAlreadyDownloaded_isNotNewOrUpdated() throws Exception {
+        File mediaDir1 = TempFiles.createTempDir();
+
+        formsRepository.save(new Form.Builder()
+                .formId("form-2")
+                .version("1")
+                .md5Hash("form-2_v1-hash")
+                .formFilePath(FormUtils.createXFormFile("form-2", "1").getAbsolutePath())
+                .formMediaPath(mediaDir1.getAbsolutePath())
+                .build());
+
+        File oldMediaFile1 = TempFiles.createTempFile(mediaDir1, "blah", ".txt");
+        writeToFile(oldMediaFile1, "old blah");
+
+        File mediaDir2 = TempFiles.createTempDir();
+
+        formsRepository.save(new Form.Builder()
+                .formId("form-2")
+                .version("2")
+                .md5Hash("form-2-hash")
+                .formFilePath(FormUtils.createXFormFile("form-2", "2").getAbsolutePath())
+                .formMediaPath(mediaDir2.getAbsolutePath())
+                .build());
+
+        File oldMediaFile2 = TempFiles.createTempFile(mediaDir2, "blah", ".txt");
```

Comment:
Could you just reference the `mediaFile` field here to make it really explicit we're using the same file?

Reply:
Ok I improved it a bit.